hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -180,7 +186,14 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
             return oldestPendingCompactionInstant
                 .map(instant -> HoodieTimeline.compareTimestamps(instant.getTimestamp(), GREATER_THAN, s.getTimestamp()))
                 .orElse(true);
-          }).limit(commitTimeline.countInstants() - minInstantsToKeep);
+          });
+      if (config.getFailedWritesCleanPolicy() == HoodieCleaningPolicy.HoodieFailedWritesCleaningPolicy.LAZY
+          && config.getMultiWriterConcurrencyModel() != MultiWriterConcurrencyModel.LEGACY_SINGLE_WRITER) {
```

Comment:
do we need to guard this by `config.getMultiWriterConcurrencyModel() != MultiWriterConcurrencyModel.LEGACY_SINGLE_WRITER` ? it should technically work right? if so, best to keep just the one flag to fence